app/views/assessments/show.html.erb

```diff
@@ -157,6 +156,10 @@
         <% end %>
       <% end %>
     </ul>
+    <% if @cud.instructor? then %>
+      <%= link_to 'Add '+@assessment.display_name+' Attachment', new_course_assessment_attachment_path(@course, @assessment) %>
+    <% end %>
+
```

Comment:
I think that you want to display this regardless of whether `@assesment.attachments.length` is greater than 0.